### PR TITLE
Orange stepbar

### PR DESCRIPTION
**Pull request by louismaximepiton**
Backport to v4 the design of Stepbars. (38px height to 30px)

Preview : [link](https://deploy-preview-1039--boosted.netlify.app/docs/4.6/components/stepbar/)

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -1404,8 +1404,10 @@ $o-scroll-up-bottom: (
 ) !default;
 
 // Step process bar
-$o-stepbar-height:      $spacer * 2 !default;
-$o-stepbar-arrow-width: 1.0625rem !default;
+$o-stepbar-padding-y:   divide(.875rem, 2) !default;
```
Comment (julien-deramond) [inline scss/_variables.scss:1407]: IMO you can directly use `.4375rem` rather than the division here.

File: scss/_o-step-bar.scss [hunk 2]
```diff
@@ -14,7 +14,7 @@
 .stepbar-item {
   display: flex;
   flex: 1;
-  padding: map-get($spacers, 2);
+  padding: $o-stepbar-padding-y map-get($spacers, 2);
```
Comment (julien-deramond) [inline scss/_o-step-bar.scss:17]: I'd use a specific var for the 2nd part:
```scss
$o-stepbar-padding-x:   map-get($spacers, 2) !default;
```

File: scss/_o-step-bar.scss [hunk 3]
```diff
@@ -54,6 +54,8 @@
   max-width: 1.25ch;
   margin: auto;
   overflow: hidden;
+  font-size: $font-size-sm;
+  line-height: $o-stepbar-line-height;
```
Comment (julien-deramond) [inline scss/_o-step-bar.scss:58]: I'd use here `line-height: $line-height-sm;` here.

Isn't there a bug with `$line-height-sm`? When I apply it, the set value is `calc(divide(16, 14))` and the browser can't use it. Small buttons seems broken as well.

File: scss/_o-step-bar.scss [hunk 4]
```diff
@@ -81,27 +83,30 @@
...
       content: "";
       background-image: escape-svg(url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 17 40'><path fill='#{$black}' d='M0 0l14 20L0 40z'/><path fill='#{$white}' d='M3 0H0l14 20L0 40h3l14-20z'/></svg>"));
       background-size: $o-stepbar-arrow-width $o-stepbar-height;
       transform: translateX(#{"/*rtl:insert:-*/"}$o-stepbar-arrow-width) #{"/*rtl:insert:rotate(180deg)*/"};
     }
 
-    &:last-child::after {
-      display: none;
+    &:first-child {
+      padding-left: divide($o-stepbar-arrow-width, 2);
```
Comment (julien-deramond) [inline scss/_o-step-bar.scss:105]: Could be a specific variable.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
Since the size of this component has changed, the visible focus is more difficult to see. Especially when the step has a black background. This is not perfect in v5 neither but a bit better. Maybe you could look at it to see what we could do about it.

File: scss/_o-step-bar.scss (hunk 3 above)
Comment (louismaximepiton) [reply scss/_o-step-bar.scss:58]: Yes it looks like a bug, I guess that it comes from lines 280 and 281 in the `_variables.scss`, the calc function is too much there. At least it works well without.

Comment (julien-deramond) [reply scss/_o-step-bar.scss]: Issue created: https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/1066

Comment (julien-deramond) [reply scss/_o-step-bar.scss]: This issue is now closed. Resolving this conversation and rebasing this PR.

File: scss/_o-step-bar.scss (hunk 4 above)
Comment (julien-deramond) [reply scss/_o-step-bar.scss]: It isn't `$o-stepbar-padding-x-xc` rather than `$o-stepbar-paddingx-xc`? And what's `xc`?

Comment (louismaximepiton) [reply scss/_o-step-bar.scss]: `xc` stands for `extreme-child`

Comment (julien-deramond) [reply scss/_o-step-bar.scss]: I don't have something better to propose so let's go with that.

**Review by julien-deramond — APPROVED** [review]
Sounds good to me :rocket:

**Pull request merged**